Home_Work_27/HW27.cpp

```diff
@@ -0,0 +1,574 @@
+
+	resultOfSum = 0;
+	begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread8(sumElements, vector100000, 0, 49999, std::ref(resultOfSum));
+	std::thread tread9(sumElements, vector100000, 50000, 99999, std::ref(resultOfSum));
+
+	tread8.join();
+	tread9.join();
+
+	end = std::chrono::high_resolution_clock::now();
+	duration = std::chrono::duration_cast<std::chrono::microseconds>(end - begin);
+	std::cout << "Sum of all elements in vector of size 100000" << std::endl;
+	std::cout << "Sum result " << resultOfSum << std::endl;
+	std::cout << "Time taken " << duration.count() << std::endl;
+
+	resultOfSum = 0;
+	begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread10(sumElements, vector1000000, 0, 499999, std::ref(resultOfSum));
+	std::thread tread11(sumElements, vector1000000, 500000, 999999, std::ref(resultOfSum));
+
+	tread10.join();
+	tread11.join();
+
+	end = std::chrono::high_resolution_clock::now();
+	duration = std::chrono::duration_cast<std::chrono::microseconds>(end - begin);
+	std::cout << "Sum of all elements in vector of size 1000000" << std::endl;
+	std::cout << "Sum result " << resultOfSum << std::endl;
+	std::cout << "Time taken " << duration.count() << std::endl;
+
+
+	resultOfSum = 0;
+	begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread12(sumElements, vector10000000, 0, 4999999, std::ref(resultOfSum));
+	std::thread tread13(sumElements, vector10000000, 5000000, 9999999, std::ref(resultOfSum));
+
+	tread12.join();
+	tread13.join();
+
+	end = std::chrono::high_resolution_clock::now();
+	duration = std::chrono::duration_cast<std::chrono::microseconds>(end - begin);
+	std::cout << "Sum of all elements in vector of size 10000000" << std::endl;
+	std::cout << "Sum result " << resultOfSum << std::endl;
+	std::cout << "Time taken " << duration.count() << std::endl;
+
+
+	resultOfSum = 0;
+	begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread14(sumElements, vector100000000, 0, 49999999, std::ref(resultOfSum));
+	std::thread tread15(sumElements, vector100000000, 50000000, 99999999, std::ref(resultOfSum));
+
+	tread14.join();
+	tread15.join();
+
+	end = std::chrono::high_resolution_clock::now();
+	duration = std::chrono::duration_cast<std::chrono::microseconds>(end - begin);
+	std::cout << "Sum of all elements in vector of size 100000000" << std::endl;
+	std::cout << "Sum result " << resultOfSum << std::endl;
+	std::cout << "Time taken " << duration.count() << std::endl;
+
+
+	std::cout << std::endl;
+	std::cout << "Eigth Threads";
+	std::cout << std::endl;
+
+	resultOfSum = 0;
+	begin = std::chrono::high_resolution_clock::now();
+
+	std::thread tread16(sumElements, vector10000, 0, 1249, std::ref(resultOfSum));
+	std::thread tread17(sumElements, vector10000, 1250, 2499, std::ref(resultOfSum));
+	std::thread tread18(sumElements, vector10000, 2500, 3749, std::ref(resultOfSum));
+	std::thread tread19(sumElements, vector10000, 3750, 4999, std::ref(resultOfSum));
```

Comment:
So much copy-paste code) I bet it can be done in much less repetitive way with just simple algorithm and several variables for threads_count, vec_size, chunk_size